mh/session.py

```diff
@@ -54,6 +54,7 @@ def __init__(self):
         self.binary_setting = b""
         self.search_payload = None
         self.hunter_info = pati.HunterSettings()
+        self.graceful_shutdown = False
```

Comment:
This can be avoided since it only happens when an exception occurs. You just need to clean it in the `except` block.

Example:
```python
    def handle(self):
        """Default PAT handler."""
        (...)
        self.sendReqConnection()
        try:
            self.handle_client()
        except Exception as e:
            traceback.print_exc()
            # Insert the cleaning logic here
            # self.leave_layer() ???
```